### PR TITLE
Refactor Jenkinsfile to leverage git-semver and optimize Dockerfile b…

### PR DESCRIPTION
**Pull request by ernestojeda**
- [x] Remove and ignore VERSION file
- [x] Add make target in Dockerfile
- [x] Remove Dockerfile.build/Dockerfile.build-arm64 as just a simple agent is needed
- [x] Implement git-semver steps

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

### CODE REVIEW

**Comment by ernestojeda** [comment]
Here is the sandbox job: https://jenkins.edgexfoundry.org/sandbox/view/All/job/app-service-configurable-verify-pipeline/

File: Jenkinsfile [hunk 1]
```diff
@@ -30,66 +33,100 @@ pipeline {
...
+                                
+                                sh 'echo Currently Building version: `cat ./VERSION`'
+
+                                script {
+                                    image_amd64 = docker.build(
+                                        'docker-app-service-configurable',
+                                        "--build-arg MAKE='test build' --label 'git_sha=${env.GIT_COMMIT}' ."
+                                    )
+
+                                    // edgeXCodecov('app-functions-sdk-go-codecov-token')
```
Comment (lenny-intel) [inline Jenkinsfile]: Remove if no longer needed. Is CodeCov enabled?

File: Jenkinsfile [hunk 2]
```diff
@@ -30,66 +33,100 @@ pipeline {
...
-                script {
-                    def semverVersion = edgeXSemver()
-                    env.setProperty('VERSION', semverVersion)
+                                script {
+                                    image_arm64 = docker.build(
+                                        'docker-app-service-configurable-arm64',
+                                        "--build-arg MAKE='test build' --label 'git_sha=${env.GIT_COMMIT}' ."
+                                    )
+
+                                    // edgeXCodecov('app-functions-sdk-go-codecov-token')
```
Comment (lenny-intel) [inline Jenkinsfile]: Remove if no longer needed. Is CodeCov enabled?

File: Jenkinsfile (hunk 1 above)
Comment (ernestojeda) [reply Jenkinsfile]: I need to check with eball to see if they have configured the token

File: Jenkinsfile (hunk 2 above)
Comment (ernestojeda) [reply Jenkinsfile]: I need to check with eball to see if they have configured the token

File: Jenkinsfile (hunk 1 above)
Comment (ernestojeda) [reply Jenkinsfile]: I will remove for now...the way we have implemented the test and build steps do not allow codecov to gather the test files needed. Plus the make test target would need to be updated to: `GO111MODULE=on go test -coverprofile=coverage.out ./...`

**Comment by ernestojeda** [comment]
@lenny-intel I updated the pipeline to support codecov. From a developer's perspective nothing really has changed. You can build the docker image using the Dockerfile at the base of the repo without anything to override. Sandbox job is here: https://jenkins.edgexfoundry.org/sandbox/view/All/job/app-service-configurable-verify-pipeline/13/

**Pull request merged**